### PR TITLE
docs(troubleshooting): WebFinger requirement for tailscale

### PR DESCRIPTION
**Pull request by mffap**
Add guidance to our troubleshoot section for webfinger support. This question came up (and people searched for it) since Tailscale requires users to setup a WebFinger endpoint when using custom oidc IDPs.

### CODE REVIEW

File: docs/docs/support/troubleshooting.md [hunk 1]
```diff
@@ -37,3 +36,11 @@ Do you still face this issue? Please contact us, and we will help you find out w
 If you're in an self-hosting scenario with a custom domain, you need to instruct ZITADEL to use the `ExternalDomain`.
```
Comment (dakshitha) [inline docs/docs/support/troubleshooting.md:36]:
```suggestion
If you're self hosting with a custom domain, you need to instruct ZITADEL to use the `ExternalDomain`.
```

File: docs/docs/support/troubleshooting.md [hunk 2]
```diff
@@ -37,3 +36,11 @@ Do you still face this issue? Please contact us, and we will help you find out w
...
 You can find more instruction in our guide about [custom domains](https://zitadel.com/docs/self-hosting/manage/custom-domain).
```
Comment (dakshitha) [inline docs/docs/support/troubleshooting.md:37]:
```suggestion
You can find further instructions in our guide about [custom domains](https://zitadel.com/docs/self-hosting/manage/custom-domain).
```

File: docs/docs/support/troubleshooting.md [hunk 3]
```diff
@@ -8,7 +8,6 @@ You will find some possible error messages here, what the problem is and what so
 Join or [Chat](https://zitadel.com/chat) or open a [Discussion](https://github.com/zitadel/zitadel/discussions).
 :::
 
-
 ## User Agent does not correspond
```
Comment (dakshitha) [inline docs/docs/support/troubleshooting.md:11]:
```suggestion
## User agent does not correspond
```

File: docs/docs/support/troubleshooting.md [hunk 4]
```diff
@@ -37,3 +36,11 @@ Do you still face this issue? Please contact us, and we will help you find out w
...
 You can find more instruction in our guide about [custom domains](https://zitadel.com/docs/self-hosting/manage/custom-domain).
 We also provide a guide on how to [configure](https://zitadel.com/docs/self-hosting/manage/configure) ZITADEL with variables from files or environment variables.
+
+## WebFinger requirement for Tailscale
+
+The WebFinger requirement and setup is a step a user has to take outside of their IdP set-up. WebFinger is a protocol which supports the ability for OIDC issuer discovery, and we use it to prove that the user has administrative control over the domain and to retrieve the issuer. This is a requirement we have in place for all users, regardless of their IdP, who use custom OIDC with Tailscale.
+
+On their custom domain, e.g example.com, they need to host a WebFinger endpoint at https://example.com/.well-known/webfinger. When queried, this endpoint returns a JSON response detailing the issuer. In your case, they would need to host the endpoint with the link to the ZITADEL issuer. We only look up this endpoint once when a user signs up, and will only look up this endpoint again if the user needs to make a configuration change to their identity provider.
```
Comment (dakshitha) [inline docs/docs/support/troubleshooting.md:44]:
```suggestion
On their custom domain, e.g example.com, users need to host a WebFinger endpoint at https://example.com/.well-known/webfinger. When queried, this endpoint returns a JSON response detailing the issuer. Users would need to host the endpoint with the link to the ZITADEL issuer. Tailscale only looks up this endpoint once when a user signs up, and will only look up this endpoint again if the user needs to make a configuration change to their identity provider.
```

**Review by dakshitha — COMMENTED** [review]
Suggested changes.

**Comment by mffap** [comment]
Added the changes, good to go?

**Review by dakshitha — APPROVED** [review]

**Pull request merged**